api/routes/user/signup.js

```diff
@@ -0,0 +1,140 @@
+import express from 'express';
+import passport from 'passport';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+
+
+// import * as UserController from '../../conrollers/users-controller.js';
+ import User from '../../models/user.js';
+import user from '../../models/user.js';
+
+
+const router = express.Router();
+
+// router.post('/signup',UserController.signup)
+
+
+// router.post('/signup', (req, res) => {
+//     console.log(req.body);
+//     User.findOne({email:req.body.email}).then((user)=>{
+        
+//         if(user){
+//             return res.status(400).json({email:'a user has already registered with this email'})
+//         }else{
+//             const newUser = new User({
+//                 name:req.body.name,
+//                 email:req.body.email
+//         });
+//         newUser.save()
+//         return res.status(200).json({msg:newUser})
+//         }
+//     })
+// })
+
+router.post('/', async (req,res)=>{
```

Comment:
I get that you are using let instead of const because the scope of the const would be just inside the try block. But nobody is stopping you from moving everything in one single try/catch block instead of having 3 try/catch blocks